### ISSUE TITLE
Integrate Electrum wallet as a client library

### ISSUE THREAD

**Issue by rkhamis** [threefoldtech/jumpscale_lib#28]
*Issue migrated from [https://api.github.com/repos/Jumpscale/lib9/issues/339](), opened by @ahussein*

We need to integrate [Electrum] (https://github.com/spesmilo/electrum) as a client library to connect to an existing electrum server.
The client library should provide similar functionality to the existing rivine client j.tools.clients.rivine
We also might need to move both rivine and the new client to a new namespace j.tools.blockchain.[rivine/electrum]

What is important to notice is that the client code in the Electrum wallet was not really made to be used an embedded library, so some investigation needs to be done. We need to look into the scripts directory and also on the main entry point [script] (https://github.com/spesmilo/electrum/blob/master/run_electrum)

**Comment by rkhamis** [comment]
*commented by @ahussein
merged to development, docs can be found here: https://github.com/Jumpscale/lib9/tree/development/JumpScale9Lib/clients/blockchain/electrum*

**Comment by rkhamis** [comment]
*commented by @despiegk
need to document more commands,
e.g. cannot see how to create a transaction*

**Comment by Pishoy** [comment]
```
electrum
commit e7ef4aa4f6fa34fc8a96dcc9def198b76a6b9b22
Merge: deee292 576c271
Author: ThomasV <thomasv@electrum.org>
Date:   Mon Aug 13 11:21:08 2018 +0200

    Merge pull request #4611 from tiagotrs/master
    
    fix Revealer hidpi related bug #4576, and make PNGs the same size
jumpscale_core
commit fdca90e9dd5c2b277f18fcadc3e82d94b3f06e29
Author: BolaNasr <bola.e.nasr@gmail.com>
Date:   Tue Aug 14 11:36:06 2018 +0200

    fix mixed usage of user profile files ( use .profile_js) #11
jumpscale_lib
commit d954725183af61f0b6fba26e5d9b88f100988e8a
Author: BolaNasr <bola.e.nasr@gmail.com>
Date:   Tue Aug 14 11:35:04 2018 +0200

    fix mixed usage of user profile files ( use .profile_js)
jumpscale_prefab
commit 6578b69ee9312df4f398524dd35dfcf3e43a1483
Author: BolaNasr <bola.e.nasr@gmail.com>
Date:   Tue Aug 14 11:32:45 2018 +0200

    fix mixed usage of user profile files ( use .profile_js)
```
```
In [21]: SEED = 'net shield travel gather west quote afraid salad spawn casino smile smoke boil flower rescue ima
    ...: ge antenna soda silent bounce husband tail square phrase'

In [22]: from electrum.mnemonic import Mnemonic
[ecc] warning: libsecp256k1 library not available, falling back to python-ecdsa

In [23]: from electrum.mnemonic import Mnemonic

In [24]: SEED = Mnemonic('n').make_seed(num_bits=256)
language n
wordlist has 2048 words
make_seed. prefix: '01' entropy: 264 bits
24 words

In [25]: ELECTRUM_DIR = '/opt/var/data/electrum'
    ...: RPC_USER = 'user'
    ...: RPC_PASS = 'pass'
    ...: RPC_PORT = 7777
    ...: RPC_HOST = 'localhost'
    ...: WALLET_NAME = 'testwallet'
    ...: 
    ...: 

In [26]: client_data = {
    ...:       'server': "{}:{}:s".format(RPC_HOST, RPC_PORT),
    ...:       'rpc_user': RPC_USER,
    ...:       'rpc_pass_': RPC_PASS,
    ...:       'seed_': SEED,
    ...:       'password_': "pass",
    ...:       "passphrase_": "",
    ...:       "electrum_path": ELECTRUM_DIR,
    ...:       "testnet": 1
    ...:   }
    ...:   

In [27]:   electrum_cl = j.clients.electrum.get(instance=WALLET_NAME,data=client_data)

In [28]: electrum_cl.config.save()

In [29]: electrum_cl
Out[29]: 
js_object:config:j.clients.electrum:testwallet

    
    electrum_path = '/opt/var/data/electrum'
    
    fee = 10000
    
    passphrase_ = ''
    
    password_ = '316131646339316339303733323563363932373164646630282022ebdfdf558c10811450b2ad1e1b7cddfeb3'
    
    rpc_pass_ = '316131646339316339303733323563363932373164646630282022ebdfdf558c10811450b2ad1e1b7cddfeb3'
    
    rpc_user = 'user'
    
    seed_ = '66613662303839646364396632343132316430333065633403563f0d3c10dbe6b24c41011894a4c325dc3d6dab94f8c9137d49243b73bc634244e78ea74c3ed91f8ec4bab06ed2172dbdba85ac7fca2e7788e0a9294
        acafb00a0f63e5d773417df54ad0f0f8c530167baf732fd6d4d8a3f04db60d10617bcb83bc9330a89201da5fe36164d3f3e7d6ef6bd0dca34fbf096e9835322d0c32aec5c59565e395060b75f55d7a54a4395a0c9aff555c
        c01e5a87801'
    
    server = 'localhost:7777:s'
    
    testnet = 1
    
    

In [30]: electrum_cl = j.clients.electrum.get(WALLET_NAME)

In [31]: electrum_cl
Out[31]: 
js_object:config:j.clients.electrum:testwallet

    
    electrum_path = '/opt/var/data/electrum'
    
    fee = 10000
    
    passphrase_ = ''
    
    password_ = '316131646339316339303733323563363932373164646630282022ebdfdf558c10811450b2ad1e1b7cddfeb3'
    
    rpc_pass_ = '316131646339316339303733323563363932373164646630282022ebdfdf558c10811450b2ad1e1b7cddfeb3'
    
    rpc_user = 'user'
    
    seed_ = '66613662303839646364396632343132316430333065633403563f0d3c10dbe6b24c41011894a4c325dc3d6dab94f8c9137d49243b73bc634244e78ea74c3ed91f8ec4bab06ed2172dbdba85ac7fca2e7788e0a9294
        acafb00a0f63e5d773417df54ad0f0f8c530167baf732fd6d4d8a3f04db60d10617bcb83bc9330a89201da5fe36164d3f3e7d6ef6bd0dca34fbf096e9835322d0c32aec5c59565e395060b75f55d7a54a4395a0c9aff555c
        c01e5a87801'
    
    server = 'localhost:7777:s'
    
    testnet = 1
    
    

[SimpleConfig] electrum directory /opt/var/data/electrum/testnet
[WalletStorage] wallet path /opt/var/data/electrum/testnet/wallets/testwallet
[WalletStorage] saved /opt/var/data/electrum/testnet/wallets/testwallet
[profiler] WalletStorage.write 0.0094
[profiler] Standard_Wallet.load_transactions 0.0000
[profiler] Standard_Wallet.load_local_history 0.0000
[profiler] Standard_Wallet.check_history 0.0000
^[[28;1R^[[28;1R^[[28;1R^[[28;1R^[[28;1R[Network] blockchains dict_keys([0])
[Network] starting network
[Network] connecting to localhost:7777:s as new interface
[localhost] [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:645)
[hsmithsxurybd7uh.onion] cannot resolve hostname
[electrumx.kekku.li] cannot resolve hostname
[testnet1.bauerj.eu] failed to connect [Errno 101] Network is unreachable
[localhost] SSL error retrieving SSL certificate: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:645)
[testnet.qtornado.com] connected
[electrum-testnet-unlimited.criptolayer.net] connected
[testnet.hsmiths.com] connected
[Network] switching to electrum-testnet-unlimited.criptolayer.net:50102:s
[Network] sending subscriptions to electrum-testnet-unlimited.criptolayer.net:50102:s 0 0
[Network] no height for main interface
[Network] switching to testnet.qtornado.com:51002:s
[Network] sending subscriptions to testnet.qtornado.com:51002:s 26 0
[Network] fee_histogram [[750, 107645], [500, 522427], [498, 437076], [495, 700], [404, 431], [390, 1536], [389, 1028], [373, 106871], [280, 215832], [195, 217797], [187, 248190], [156, 281504]]
[Network] fee_estimates[25] 124870
[Network] fee_estimates[10] 218533
[Network] fee_estimates[5] 283770
[Network] fee_estimates[2] 500446
[Network] relayfee 2606
[Network] fee_histogram [[500, 124331], [375, 86412], [187, 330271], [186, 265256], [168, 225], [157, 419], [101, 156703]]
[WalletStorage] saved /opt/var/data/electrum/testnet/wallets/testwallet
[profiler] WalletStorage.write 0.0072
[profiler] Standard_Wallet.save_transactions 0.0076
[WalletStorage] saved /opt/var/data/electrum/testnet/wallets/testwallet
[profiler] WalletStorage.write 0.0079
[Network] fee_estimates[25] 124865
[Network] relayfee 1000
[Network] fee_estimates[10] 200502
[Network] fee_estimates[5] 250234
[Network] fee_estimates[2] 471716
[profiler] Standard_Wallet.save_transactions 0.0001
[WalletStorage] saved /opt/var/data/electrum/testnet/wallets/testwallet
[profiler] WalletStorage.write 0.0069
[profiler] WalletStorage.write 0.0000
[Network] stopping network
[Network] stopped
In [32]: electrum_cl.wallet.listaddresses()
Out[32]: 
['n2nzNok5w4A68yQ6cHFGjGyJr8frM5KW9L',
 'n3rG3KSkswMb3oUJrrye8Z3isoubNPwL62',
 'mnNgUk3vDgsZwaqvNwYXjHzC6oUjxU9Cbc',
 'n15HcUDGWQm8Zc7S5eck6zqJrrKdESgQqi',
 'mjzVTx9rLbt4vD9QN6jDZ7DXPoXQLp2LaB',
 'my68UNtkjpoRrgzzT1w6caxcmpoSANRHTc',
 'mhTrDyvij1jyuwLNtMNQAgVsxqLf1FhsJ3',
 'mikGPmX7R9ysnVjp29vNaYiwHMBaHdbanM',
 'mkMHRgRKEMxiJBAWHSYaUfDPPMUFsPdDmv',
 'mheEXZimcWrqm2rxuPZMbzbSwutzUpvRUi',
 'n2bgtjbQdAfcU7rgT3Q7B6r9ehGPoJGpZt',
 'mzJ3WXHsG8fRcqvBB4pS5HQHYo11TzfxCd',
 'mjd1HhZRpjDnqderJsAxHN5zbFBMyeZUrL',
 'mpGz6hjMwWQDDemW9fm9VsRjMqeDrz549A',
 'mrME6zuLFhB8N7u4utVNAQEp6D1Vqdwd3X',
 'miJGAZb5gPv1x9P2mNkBDbCUicEdB5bgHk',
 'myjQs4K2zFgqJTtxcAnpiURdf3vX7Wnc8q',
 'mxRCSovtRDhunVhgACCmt38QSTnApPX2uz',
 'mrW2nyWAKx8G5rEUWj5VqExjfCuZhuASEQ',
 'mpMHsEc6sLumLdZ2R83EbDT1gjL7hYi6XH',
 'mxTSzKwNs5LGyzAm4D4yRQUh7ye6Y66mt7',
 'msdHQJbV3Wmfn4SXyo2CvgHv33xADWBiSF',
 'n38shXi6ofUqyexhwJodJL4seZmKjXYysB',
 'moMErHWRpdypKPCigjcKZ7n358sMfysmGb',
 'n4gttTTAk2cN1YCXEQj8EabYDZdrGoN9XH',
 'moapbjFLmFxpapJTQvVi35Y9aSfAbRWv6Q']

In [33]: electrum_cl.wallet.getbalance()
Out[33]: {'confirmed': '0'}

In [34]:  j.tools.prefab.local.runtimes.golang.config()
```
